test/ymlr/encode_test.exs

```diff
@@ -49,6 +49,7 @@ defmodule Ymlr.EncodeTest do
     end
 
     test "quoted strings - starts with special char" do
+      assert_identity_and_output("?", ~S('?'))
       assert_identity_and_output("!tag", ~S('!tag'))
```

Comment:
Reading the documentation you liked, I think we should also add tests for the other indicator characters (ie. the this list: `[22] c-indicator = `. Also, I would put them in a separate `test` block: Would you add that to your PR? Otherwise we can merge and I'll do it.


```suggestion
    test "quoted strings - indicator characters on their own" do
      assert_identity_and_output("-", ~S('-'))
      assert_identity_and_output("?", ~S('?'))
      assert_identity_and_output(":", ~S(':'))
      assert_identity_and_output("[", ~S('['))
      assert_identity_and_output("]", ~S(']'))
      assert_identity_and_output("{", ~S('{'))
      assert_identity_and_output("}", ~S('}'))
      assert_identity_and_output("#", ~S('#'))
      assert_identity_and_output("&", ~S('&'))
      assert_identity_and_output("&", ~S('&'))
      assert_identity_and_output("*", ~S('*'))
      assert_identity_and_output("*", ~S('*'))
      assert_identity_and_output("!", ~S('!'))
      assert_identity_and_output("!", ~S('!'))
      assert_identity_and_output("|", ~S('|'))
      assert_identity_and_output("|", ~S('|'))
      assert_identity_and_output(">", ~S('>'))
      assert_identity_and_output(">", ~S('>'))
      assert_identity_and_output("%", ~S(%))
      assert_identity_and_output("@", ~S('@'))
      assert_identity_and_output("`", ~S('`'))
    end
    
    test "quoted strings - starts with special char" do
      assert_identity_and_output("!tag", ~S('!tag'))
```